tests/test_device.py

```diff
@@ -62,8 +63,61 @@ def test_supported(self):
             self.assertTrue(len(ops) > 0)
             self.assertTrue(len(exps) > 0)
 
-            for op in ops.union(exps):
-                self.assertTrue(dev.supported(op))
+            for op in ops:
+                self.assertTrue(dev.supports_operation(op))
+
+            for obs in exps:
+                self.assertTrue(dev.supports_observable(obs))
+
+    @patch.multiple(Device, __abstractmethods__=set(), operations=PropertyMock(return_value=['PauliX']))
+    def test_supports_operation_argument_types(self):
+        """check that a the different argument types for the function
+           device.supports_operation are supported"""
```

Comment:
Might say something along the lines of "Checks that device.supports_operations returns the correct result when passed both string and Operation arguments"